distributed/worker.py

```diff
@@ -2759,18 +2767,37 @@ def get_client(address=None, timeout=None, resolve_address=True) -> Client:
         pass
     else:
         if not address or worker.scheduler.address == address:
-            return worker._get_client(timeout=timeout)
-
-    from distributed.client import Client
+            client = worker._get_client(timeout=timeout)
+            if asynchronous is None or client._asynchronous == asynchronous:
```

Comment:
I don't think it matters what mode the client was created in, but what client.sync will do when it's called, so you do want the `client.asynchronous` flag